runtime/gateway.go

```diff
@@ -353,11 +356,19 @@ func (gateway *Gateway) setupHTTPServer() error {
 }
 
 func (gateway *Gateway) setupTChannel(config *StaticConfig) error {
+	// TODO: (lu) channel name and opts
+	ch, err := tchannel.NewChannel("Gateway", nil)
```

Comment:
That's not really what I meant. This is duplicate with `runtime/tchannel_server.go:56`

we should use `tchannelServer.Channel`.

Never allocate two channels.